src/com/projectkorra/projectkorra/airbending/flight/FlightMultiAbility.java

```diff
@@ -19,22 +19,29 @@
 import com.projectkorra.projectkorra.Element;
 import com.projectkorra.projectkorra.GeneralMethods;
 import com.projectkorra.projectkorra.ProjectKorra;
+import com.projectkorra.projectkorra.ability.CoreAbility;
 import com.projectkorra.projectkorra.ability.FlightAbility;
 import com.projectkorra.projectkorra.ability.MultiAbility;
 import com.projectkorra.projectkorra.ability.util.MultiAbilityManager;
 import com.projectkorra.projectkorra.ability.util.MultiAbilityManager.MultiAbilityInfoSub;
+import com.projectkorra.projectkorra.airbending.AirScooter;
+import com.projectkorra.projectkorra.airbending.AirSpout;
+import com.projectkorra.projectkorra.firebending.FireJet;
 import com.projectkorra.projectkorra.util.ActionBar;
 import com.projectkorra.projectkorra.util.DamageHandler;
+import com.projectkorra.projectkorra.util.FlightHandler.Flight;
 import com.projectkorra.projectkorra.util.MovementHandler;
 import com.projectkorra.projectkorra.util.ParticleEffect;
+import com.projectkorra.projectkorra.waterbending.WaterSpout;
 
 public class FlightMultiAbility extends FlightAbility implements MultiAbility{
 	
 	public static Map<UUID, UUID> requestedMap = new HashMap<>();
 	public static Map<UUID, Long> requestTime = new HashMap<>();
 	
 	private static Set<UUID> flying = new HashSet<>();
-	private boolean canFly, hadFly, hadGlide;
+	private boolean hadGlide;
+	private Flight flight;
```

Comment:
You should never store an instance of Flight yourself. FlightHandler will manage all of this.

Reply:
In the FlightMultiAbility I set the states of flying various times depending on the mode. In the ending mode, I want them to have whatever the state of flying was before using the ability or starting to fly. I store it so I can access those stored states whenever they go to ending mode (since it's a mutliability, they can switch between binds whenever, need to check every iteration of progress). I do this because FlightHandler doesn't manage it in the way I said, not when I need it to. So yes I do need to store the instance of Flight, or else why does the createInstance method return Flight at all, if I'm not intended to store it?

Reply:
> In the FlightMultiAbility I set the states of flying various times depending on the mode.
That's fine, though you shouldn't be doing that from the Flight instance. FlightHandler works by storing a player's flight state for multiple identifiers and does not reverting until all instances have been removed. If you were want to give them flight, use the createInstance method and then do whatever you want with the player (setFlying(true/false), setAllowFlight(true/false)). Then when you're done, call the removeInstance method. If they have no other instances of Flight outstanding, and only if that is true, it will revert their flight state back to what it was when you (or their first flight instance) called createFlight. If you are manually setting their flight back when their FlightMultiAbility is removed, that will also prematurely revert their flight for another ability.
Eg. A player is hit by AirSwipe, Tornado, AirBlast, etc. just before their FlightMultiAbility ends. Keep in mind that the whole point of FlightHandler, and the reason we had so many fly glitches before, was because abilities did not communicate player's flight states correctly. FlightHandler is designed to save a player's fly state, and then when ALL abilities are done using flight, revert it back to what it was. You're completely defeating the purpose of using FlightHandler by manually reverting it.
> or else why does the createInstance method return Flight at all, if I'm not intended to store it?
It doesn't actually need to return the Flight instance, and I'll change that in a future PR.

Reply:
> If you are manually setting their flight back when their FlightMultiAbility is removed, that will also prematurely revert their flight for another ability.
This doesn't make sense. It shouldn't revert flight for other abilities if I just change their flying states, even if it is back to what it was before the flight. Nowhere would the code cause other abilities flight to prematurely revert.
> It doesn't actually need to return the Flight instance, and I'll change that in a future PR.
I think this should be left how it is, because it is useful to access flight (since you can't actually change anything about the flight itself)

Reply:
> It shouldn't revert flight for other abilities if I just change their flying states, even if it is back to what it was before the flight. Nowhere would the code cause other abilities flight to prematurely revert.

If another ability has someone set flying and you turn it off, or vice verse, you are prematurely reverting that abilities flight.

Reply:
And it should be? FlightMultiAbility should take priority

Reply:
No it shouldn't. If someone is hit with AirBlast a moment before ending their FlightMultiAbility, doing what you're doing will also remove AirBlast's Flight instance, likely causing the player to get kicked for flying.